clients/drcachesim/common/options.cpp

```diff
@@ -714,6 +714,18 @@ droption_t<std::string>
                    "Cache hierarchy configuration file",
                    "The full path to the cache hierarchy configuration file.");
 
+droption_t<uint64_t> op_noise_generator_add(
+    DROPTION_SCOPE_FRONTEND, "noise_generator_add", false, "Enables noise generators.",
+    "Adds the specified number of noise generators, which enables the scheduler to "
+    "interleave trace records with synthetic records. Each noise generator behaves like "
+    "a thread in its own process.");
+
+droption_t<uint64_t> op_noise_generator_num_records(
+    DROPTION_SCOPE_FRONTEND, "noise_generator_num_records", 0,
```

Comment:
So a 0 default means `-noise_generator_add 4` emits header records and then just stops?  Seems like there should be a more reasonable default; or else the _add option needs to explain that it's broken unless you also set this other option.

Reply:
No, a `0` means the 4 noise generators will return a `nullptr` as `trace_entry_t`, so no records generated.
`-noise_generator_num_records` does not count `TRACE_TYPE_THREAD` and `TRACE_TYPE_PID`, but it does count `TRACE_TYPE_THREAD_EXIT`.
So, a `1` would generate the header records + exit, but no "real" records (instructions, reads, writes).
A `2` would generate the header records + one record + exit, and so on.

I can add the fact that `-noise_generator_add` needs `-noise_generator_num_records` to actually generate noise in the option description.

Reply:
A reasonable default that generates some records seems better; if you really want to keep it 0 then yes the other option should explain the extra work the user has to do.

Reply:
Default num_records changed to 1000 for now.